.github/workflows/publish-on-tagging.yaml

```diff
@@ -2,16 +2,39 @@ name: Publish Charts on tagging
 
 on:
   push:
-    tags: '*'
+    tags: 'v*'
 
 jobs:
   publish:
     runs-on: ubuntu-latest
+    env:
+      TAG: ${{ github.ref_name }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+      - name: Install helm
```

Comment:
I think overall the code looks good to me, however I found [setup-helm](https://github.com/Azure/setup-helm), not sure if we can use it?